Frameworks/Foundation/NSException.mm

```diff
@@ -216,7 +244,18 @@ - (HRESULT)_hresult {
         return [hresultValue unsignedIntValue];
     }
 
-    TraceWarning(L"NSException", L"Called -hresult on an NSException without an HRESULT!");
-    return E_UNEXPECTED;
+    return [[self class] _HRESULTForExceptionName:self.name];
+}
+
+- (void)_processException {
+    HRESULT hr = [self _hresult];
+    HSTRING_HEADER header;
+    HSTRING hsReason = nullptr;
+    
+    if(self.reason != nil) {
+        LPCWSTR reason = (LPCWSTR)([[self reason] cStringUsingEncoding:NSUnicodeStringEncoding]);
+        WindowsCreateStringReference(reason, wcslen(reason), &header, &hsReason);
```

Comment:
reason is the backing store for hsReason and gets freed up when the autorelase pool is drained.  That is my understanding.


---
In reply to: [93297284](https://github.com/Microsoft/WinObjC/pull/1570#discussion_r93297284) [](ancestors = 93297284,93125423)